src/fstab-generator/fstab-generator.c

```diff
@@ -936,6 +936,10 @@ static int parse_fstab_one(
                         mount_is_network(fstype, options) ? SPECIAL_REMOTE_FS_TARGET :
                                                             SPECIAL_LOCAL_FS_TARGET;
 
+        /* Setting nofail for /sysroot/usr doesn't make sense so let's just silently ignore it. */
+        if (is_sysroot_usr && (flags & MOUNT_NOFAIL))
```

Comment:
FLAGS_SET (or actually you can drop the second condition completely)

Reply:
I'd rather keep the condition so I think think that the intent is then tiny bit clearer